shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java

```diff
@@ -62,38 +65,49 @@ public final class InsertStatementContext extends CommonSQLStatementContext<Inse
     
     private List<InsertValueContext> insertValueContexts;
     
+    private Map<Integer, List<InsertValueContext>> insertValueContextsMap = new LinkedHashMap<>();
+    
     private InsertSelectContext insertSelectContext;
     
     private OnDuplicateUpdateContext onDuplicateKeyUpdateValueContext;
     
-    private GeneratedKeyContext generatedKeyContext;
+    private Map<Integer, GeneratedKeyContext> generatedKeyContexts = new LinkedHashMap<>();
     
     public InsertStatementContext(final Map<String, ShardingSphereMetaData> metaDataMap, final List<Object> parameters,
                                   final InsertStatement sqlStatement, final String defaultDatabaseName) {
         super(sqlStatement);
         this.metaDataMap = metaDataMap;
         this.defaultDatabaseName = defaultDatabaseName;
         insertColumnNames = getInsertColumnNames();
-        valueExpressions = getAllValueExpressions(sqlStatement);
-        AtomicInteger parametersOffset = new AtomicInteger(0);
-        insertValueContexts = getInsertValueContexts(parameters, parametersOffset, valueExpressions);
-        insertSelectContext = getInsertSelectContext(metaDataMap, parameters, parametersOffset, defaultDatabaseName).orElse(null);
-        onDuplicateKeyUpdateValueContext = getOnDuplicateKeyUpdateValueContext(parameters, parametersOffset).orElse(null);
         tablesContext = new TablesContext(getAllSimpleTableSegments(), getDatabaseType());
         ShardingSphereSchema schema = getSchema(metaDataMap, defaultDatabaseName);
-        columnNames = useDefaultColumns() ? schema.getAllColumnNames(sqlStatement.getTable().getTableName().getIdentifier().getValue()) : insertColumnNames;
-        generatedKeyContext = new GeneratedKeyContextEngine(sqlStatement, schema)
-                .createGenerateKeyContext(insertColumnNames, getAllValueExpressions(sqlStatement), parameters).orElse(null);
+        AtomicInteger parametersOffset = new AtomicInteger(0);
+        List<InsertStatement> insertStatements = getInsertStatements();
+        for (int index = 0; index < insertStatements.size(); index++) {
+            InsertStatement insertStatement = insertStatements.get(index);
+            List<List<ExpressionSegment>> valueExpression = getAllValueExpressions(insertStatement);
+            valueExpressions.put(index, valueExpression);
+            insertValueContextsMap.put(index, getInsertValueContexts(parameters, parametersOffset, valueExpression));
+            columnNamesMap.put(index, useDefaultColumns() ? schema.getAllColumnNames(insertStatement.getTable().getTableName().getIdentifier().getValue()) : insertColumnNames);
+            generatedKeyContexts.put(index, new GeneratedKeyContextEngine(insertStatement, schema)
+                    .createGenerateKeyContext(insertColumnNames, getAllValueExpressions(insertStatement), parameters).orElse(null));
+        }
+        insertValueContexts = insertValueContextsMap.get(0);
```

Comment:
@cheese8 Can we merge insertValueContexts and insertValueContextsMap, they appear to be the same use.